This patch remedies a ModemManager abort when trying to connect a Sierra
Wireless MC7455/MC7430 with no antennas. The observed error is:

  ERROR:mm-broadband-modem.c:9359:enable: code should not be reached killed
  because of sig - 6

The abort occurs when performing an mmcli modem enable command when the modem is
in state UNKNOWN. It gets into this state (sometimes) after trying to connect a
bearer when the modem cannot properly communicate with the cellular network
(i.e. there are no antennas connected).

The problem is that after a failed bearer connect,
`get_current_consolidated_state()` returns the default value that is passed in
because the subsystems array is empty (see the implementation of
`get_current_consolidated_state`). It is not currently known why there are no
subsystems available at that point.

This patch changes the default state returned by
`get_current_consolidated_state()` from UNKNOWN to DISABLED. ModemManager can
continue working in this state.

--- ModemManager/src/mm-iface-modem.c	2018-11-20 14:31:47.166319203 +1000
+++ ModemManager/src/mm-iface-modem.c.new	2018-11-20 14:30:45.190319208 +1000
@@ -463,7 +463,7 @@
             new_state = MM_MODEM_STATE_DISCONNECTING;
             break;
         case MM_BEARER_STATUS_DISCONNECTED:
-            new_state = get_current_consolidated_state (self, MM_MODEM_STATE_UNKNOWN);
+            new_state = get_current_consolidated_state (self, MM_MODEM_STATE_DISABLED);
             break;
         }
 
